### PR TITLE
Allow anonymous voting if desired

### LINKED ISSUE #231: Allow unregistered voting on section comments if desired

### PR DESCRIPTION
**Pull request by Rikuoja**
Closes #231

### CODE REVIEW

File: democracy/models/comment.py [hunk 1]
```diff
@@ -24,11 +24,10 @@ class BaseComment(BaseModel):
...
     )
-    n_legacy_votes = models.IntegerField(
-        verbose_name=_('legacy vote count'),
-        help_text=_('number of votes imported from legacy system'),
-        default=0,
-        editable=False
+    n_unregistered_votes = models.IntegerField(
+        verbose_name=_('unregistered vote count'),
+        help_text=_('number of unregistered votes'),
+        default=0
```
Comment (akx) [inline democracy/models/comment.py]: Why was the editable flag dropped? I don't think this should be editable by humans. ([The `editable` flag does not make a field read-only.](https://docs.djangoproject.com/en/1.10/ref/models/fields/#editable))

File: democracy/views/comment.py [hunk 2]
```diff
@@ -137,22 +148,27 @@ def perform_update(self, serializer):
...
-        comment.recache_n_votes()
-        # return success
-        return response.Response({'status': 'Vote has been added'}, status=status.HTTP_201_CREATED)
+        if not request.user.is_authenticated():
+            # If the check went through, anonymous voting is allowed
+            comment.n_unregistered_votes += 1
+            comment.recache_n_votes()
+            return response.Response({'status': 'Vote has been counted'}, status=status.HTTP_200_OK)
+        else:
+            # Check if user voted already. If yes, return 400.
```
Comment (akx) [inline democracy/views/comment.py]: The comment lies. (304 is returned, not 400.)

File: democracy/views/comment.py [hunk 3]
```diff
@@ -137,22 +148,27 @@ def perform_update(self, serializer):
...
-        # update number of votes
-        comment.recache_n_votes()
-        # return success
-        return response.Response({'status': 'Vote has been added'}, status=status.HTTP_201_CREATED)
+        if not request.user.is_authenticated():
+            # If the check went through, anonymous voting is allowed
+            comment.n_unregistered_votes += 1
+            comment.recache_n_votes()
+            return response.Response({'status': 'Vote has been counted'}, status=status.HTTP_200_OK)
+        else:
```
Comment (akx) [inline democracy/views/comment.py]: Unnecessary `else:`; the previous branch has a `return`. (Removing the `else:` will also have the pleasant side effect of shortening the diff, as the rest of the function remains untouched.)

File: democracy/models/comment.py (hunk 1 above)
Comment (Rikuoja) [inline democracy/models/comment.py]: Ah, I did indeed mistake one for the other.

**Comment by Rikuoja** [comment]
@akx Fixed

**Pull request merged**